### ISSUE TITLE
fits 1.0.7 Forge package install fails under Octave.app 4.4.1

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app#150]
```
>> pkg install -forge fits
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
Package cfitsio was not found in the pkg-config search path.
Perhaps you should add the directory containing `cfitsio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cfitsio' found
save_fits_image.ccread_fits_image.cc::save_fits_image_multi_ext.cc2323:::101023:::  10: fatal errorfatal error: : fatal error: 'fitsio.h''fitsio.h'  'fitsio.h'filefile   filenotnot   notfoundfound found


#include "fitsio.h"#include "fitsio.h"

         ^~~~~~~~~~
         ^~~~~~~~~~
#include "fitsio.h"
         ^~~~~~~~~~
1 error generated.
1 error generated.
1 error generated.
make: *** [save_fits_image.oct] Error 1
make: *** Waiting for unfinished jobs....
make: *** [read_fits_image.oct] Error 1
make: *** [save_fits_image_multi_ext.oct] Error 1
mkoctfile --verbose -Wall  read_fits_image.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave/.. -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave -I/usr/local/Cellar/octave/5.1.0_6/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include/darwin  -Wall   read_fits_image.cc -o /var/tmp//oct-jtK4Pe.o
mkoctfile --verbose -Wall  save_fits_image.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave/.. -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave -I/usr/local/Cellar/octave/5.1.0_6/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include/darwin  -Wall   save_fits_image.cc -o /var/tmp//oct-OYk62F.o
mkoctfile --verbose -Wall  save_fits_image_multi_ext.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave/.. -I/usr/local/Cellar/octave/5.1.0_6/include/octave-5.1.0/octave -I/usr/local/Cellar/octave/5.1.0_6/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-12.0.1.jdk/Contents/Home/include/darwin  -Wall   save_fits_image_multi_ext.cc -o /var/tmp//oct-OxbqoS.o

pkg: error running `make' for the fits package.
error: called from
    configure_make at line 99 column 9
    install at line 184 column 7
    pkg at line 437 column 9
>>
```

**Comment by apjanke** [comment]
I have `cfitsio` installed on my machine via Homebrew now; maybe that's making a difference.

Getting a different build failure, under both Octave.app 4.4.1 and vanilla Homebrewed Octave 5.2.0. Still fits 1.0.7.

```
>> pkg install -forge fits
read_fits_image.cc:197:10: error: use of undeclared identifier 'D_NINT'
    if( (D_NINT( val ) !=  val) || (val < 0) )
         ^
save_fits_image.cc:84:12: error: use of undeclared identifier 'D_NINT'
      if( (D_NINT( val ) ==  val) )
           ^
save_fits_image_multi_ext.cc:81:12: error: use of undeclared identifier 'D_NINT'
      if( (D_NINT( val ) ==  val) )
           ^
1 error generated.
1 error generated.
make: *** [save_fits_image.oct] Error 1
make: *** Waiting for unfinished jobs....
1 error generated.
make: *** [read_fits_image.oct] Error 1
make: *** [save_fits_image_multi_ext.oct] Error 1
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/3.470/include -L/usr/local/Cellar/cfitsio/3.470/lib -lcfitsio -lpthread read_fits_image.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave/.. -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave -I/usr/local/Cellar/octave/5.2.0/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include/darwin  -Wall  -I/usr/local/Cellar/cfitsio/3.470/include  read_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-0xWWIX.o
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/3.470/include -L/usr/local/Cellar/cfitsio/3.470/lib -lcfitsio -lpthread save_fits_image.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave/.. -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave -I/usr/local/Cellar/octave/5.2.0/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include/darwin  -Wall  -I/usr/local/Cellar/cfitsio/3.470/include  save_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-sHhxxK.o
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/3.470/include -L/usr/local/Cellar/cfitsio/3.470/lib -lcfitsio -lpthread save_fits_image_multi_ext.cc
clang++ -std=gnu++11 -c -I/usr/local/opt/qt/include -fPIC -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave/.. -I/usr/local/Cellar/octave/5.2.0/include/octave-5.2.0/octave -I/usr/local/Cellar/octave/5.2.0/include  -D_THREAD_SAFE -pthread -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/adoptopenjdk-13.0.2.jdk/Contents/Home/include/darwin  -Wall  -I/usr/local/Cellar/cfitsio/3.470/include  save_fits_image_multi_ext.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-K2MxKV.o

pkg: error running `make' for the fits package.
error: called from
    configure_make at line 99 column 9
    install at line 184 column 7
    pkg at line 437 column 9
>>
```

**Comment by apjanke** [comment]
Submitted upstream: https://savannah.gnu.org/bugs/?57818

**Comment by apjanke** [comment]
Still broken under Octave.app 8.4.0 and 9.2.0 beta1. But I just don't care. Plus, it's also broken under vanilla Homebrewed octave 9.2.0. Closing as Upstream/Wontfix.

```
>> ver
----------------------------------------------------------------------
GNU Octave Version: 8.4.0 (hg id: 78c13a2594f3 + patches)
Octave.app Version: 8.4.0
GNU Octave License: GNU General Public License
Operating System: Darwin 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:09:52 PDT 2024; root:xnu-10063.121.3~5/RELEASE_X86_64 x86_64
----------------------------------------------------------------------
Package Name         | Version | Installation directory
---------------------+---------+-----------------------
            control  |   4.0.1 | /Users/janke/Library/Application Support/Octave.app/8.4.0/pkg/control-4.0.1
fuzzy-logic-toolkit  |   0.6.0 | /Users/janke/Library/Application Support/Octave.app/8.4.0/pkg/fuzzy-logic-toolkit-0.6.0
      miscellaneous  |   1.3.0 | /Users/janke/Library/Application Support/Octave.app/8.4.0/pkg/miscellaneous-1.3.0
             signal  |   1.4.5 | /Users/janke/Library/Application Support/Octave.app/8.4.0/pkg/signal-1.4.5
>>
>> pkg install -forge fits
save_fits_image.cc:84:12: error: use of undeclared identifier 'D_NINT'
      if( (D_NINT( val ) ==  val) )
           ^
save_fits_image.cc:132:47: warning: 'fortran_vec' is deprecated [-Wdeprecated-declarations]
  double * datap = const_cast<double*>( image.fortran_vec() );
                                              ^
/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/Array.h:667:3: note: 'fortran_vec' has been
      explicitly marked deprecated here
  OCTAVE_DEPRECATED (7, "for read-only access, use 'data' method instead")
  ^
/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/oct-conf-post-public.h:50:57: note:
      expanded from macro 'OCTAVE_DEPRECATED'
#    define OCTAVE_DEPRECATED(ver, msg) __attribute__ ((__deprecated__))
                                                        ^
save_fits_image_multi_ext.cc:81:12: read_fits_image.cc:197error: :10: error: use useof  undeclared identifier 'D_NINT'of
 undeclared identifier 'D_NINT'
    if( (D_NINT( val ) !=  val) || (val < 0) )
         ^
      if( (D_NINT( val ) ==  val) )
           ^
save_fits_image_multi_ext.cc:140:49: warning: 'fortran_vec' is deprecated [-Wdeprecated-declarations]
    double * datap = const_cast<double*>( image.fortran_vec() );
                                                ^
/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/Array.h:667:3: note: 'fortran_vec' has been
      explicitly marked deprecated here
  OCTAVE_DEPRECATED (7, "for read-only access, use 'data' method instead")
  ^
/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/oct-conf-post-public.h:50:57: note:
      expanded from macro 'OCTAVE_DEPRECATED'
#    define OCTAVE_DEPRECATED(ver, msg) __attribute__ ((__deprecated__))
                                                        ^
1 warning and 1 error generated.
1 error generated.
make: *** [save_fits_image.oct] Error 1
make: *** Waiting for unfinished jobs....
make: *** [read_fits_image.oct] Error 1
1 warning and 1 error generated.
make: *** [save_fits_image_multi_ext.oct] Error 1
mkoctfile --verbose -Wall -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include -L/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/lib -lcfitsio read_fits_image.cc
clang++ -std=gnu++14 -c -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qt-octapp_5/include -fPIC -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/.. -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include  -pthread -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/sundials/include -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qscintilla2-octapp-qt5/include  -Wall  -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include  read_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-B6pmiv.o
mkoctfile --verbose -Wall -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include -L/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/lib -lcfitsio save_fits_image.cc
clang++ -std=gnu++14 -c -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qt-octapp_5/include -fPIC -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/.. -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include  -pthread -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/sundials/include -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qscintilla2-octapp-qt5/include  -Wall  -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include  save_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-xc6UH4.o
mkoctfile --verbose -Wall -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include -L/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/lib -lcfitsio save_fits_image_multi_ext.cc
clang++ -std=gnu++14 -c -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qt-octapp_5/include -fPIC -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave/.. -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include/octave-8.4.0/octave -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/octave-octapp@8.4.0/8.4.0/include  -pthread -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/sundials/include -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/opt/qscintilla2-octapp-qt5/include  -Wall  -I/Applications/Octave-8.4.0.app/Contents/Resources/usr/Cellar/cfitsio/4.4.0/include  save_fits_image_multi_ext.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-XOtpYy.o

error: pkg: error running 'make' for the fits package
error: called from
    configure_make at line 117 column 9
    install at line 202 column 7
    pkg at line 619 column 9
>>
```

```
octave:27> ver
----------------------------------------------------------------------
GNU Octave Version: 9.2.0 (hg id: f7f956555e4a)
GNU Octave License: GNU General Public License
Operating System: Darwin 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:09:52 PDT 2024; root:xnu-10063.121.3~5/RELEASE_X86_64 x86_64
----------------------------------------------------------------------
Package Name   | Version | Installation directory
---------------+---------+-----------------------
      control  |   4.0.1 | /Users/janke/.local/share/octave/api-v59/packages/control-4.0.1
      doctest *|   0.8.0 | /Users/janke/.local/share/octave/api-v59/packages/doctest-0.8.0
        image  |  2.14.0 | /Users/janke/.local/share/octave/api-v59/packages/image-2.14.0
miscellaneous *|   1.3.0 | /Users/janke/.local/share/octave/api-v59/packages/miscellaneous-1.3.0
       signal  |   1.4.5 | /Users/janke/.local/share/octave/api-v59/packages/signal-1.4.5
      testify *|   0.3.3 | /Users/janke/.local/share/octave/api-v59/packages/testify-0.3.3
octave:28> pkg install -forge fits
In file included from In file included from save_fits_image.ccsave_fits_image_multi_ext.cc::1919:
:
In file included from In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/../octave/oct.h/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/../octave/oct.h::3535:
:
In file included from In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/defun-dld.h/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/defun-dld.h::3535:
:
[...]
save_fits_image_multi_ext.cc:81:12: error: use of undeclared identifier 'D_NINT'
      if( (D_NINT( val ) ==  val) )
           ^
save_fits_image.cc:84:12: error: use of undeclared identifier 'D_NINT'
      if( (D_NINT( val ) ==  val) )
           ^
save_fits_image_multi_ext.cc:140:43: error: 'this' argument to member function 'fortran_vec' has type 'const NDArray', but
      function is not marked const
    double * datap = const_cast<double*>( image.fortran_vec() );
                                          ^~~~~
/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/Array.h:666:20: note: 'fortran_vec' declared here
  OCTARRAY_API T * fortran_vec ();
                   ^
save_fits_image.cc:132:41: error: 'this' argument to member function 'fortran_vec' has type 'const NDArray', but function is not
      marked const
  double * datap = const_cast<double*>( image.fortran_vec() );
                                        ^~~~~
/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/Array.h:666:20: note: 'fortran_vec' declared here
  OCTARRAY_API T * fortran_vec ();
                   ^
read_fits_image.cc:197:10: error: use of undeclared identifier 'D_NINT'
    if( (D_NINT( val ) !=  val) || (val < 0) )
         ^
3 warnings and 2 errors generated.
3 warnings and 2 errors generated.
make: *** [save_fits_image_multi_ext.oct] Error 1
make: *** Waiting for unfinished jobs....
make: *** [save_fits_image.oct] Error 1
3 warnings and 1 error generated.
make: *** [read_fits_image.oct] Error 1
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/4.4.1/include -L/usr/local/Cellar/cfitsio/4.4.1/lib -lcfitsio read_fits_image.cc
clang++ -std=gnu++17 -c  -fPIC -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/.. -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave -I/usr/local/Cellar/octave/9.2.0/include  -pthread -g -O2  -Wall  -I/usr/local/Cellar/cfitsio/4.4.1/include  read_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-IGbKsp.o
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/4.4.1/include -L/usr/local/Cellar/cfitsio/4.4.1/lib -lcfitsio save_fits_image.cc
clang++ -std=gnu++17 -c  -fPIC -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/.. -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave -I/usr/local/Cellar/octave/9.2.0/include  -pthread -g -O2  -Wall  -I/usr/local/Cellar/cfitsio/4.4.1/include  save_fits_image.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-L9ruJc.o
mkoctfile --verbose -Wall -I/usr/local/Cellar/cfitsio/4.4.1/include -L/usr/local/Cellar/cfitsio/4.4.1/lib -lcfitsio save_fits_image_multi_ext.cc
clang++ -std=gnu++17 -c  -fPIC -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/.. -I/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave -I/usr/local/Cellar/octave/9.2.0/include  -pthread -g -O2  -Wall  -I/usr/local/Cellar/cfitsio/4.4.1/include  save_fits_image_multi_ext.cc -o /var/folders/_4/9mx5ryp52bb_z6drbcbrhwl40000gn/T//oct-pONJHZ.o

error: pkg: error running 'make' for the fits package
error: called from
    configure_make at line 117 column 9
    install at line 202 column 7
    pkg at line 619 column 9
octave:29>
```
